docs/design/multi_slo_doc_support.md

```diff
@@ -0,0 +1,102 @@
+
+The service 'ocm' [has **two** SLO-documents](https://gitlab.cee.redhat.com/service/app-interface/-/tree/1b368998a913be3f2d45a13934eb6403d4e751de/data/services/ocm/slo-documents). Each of these documents define SLOs with the same set of names (latency, errors, availability). There are a total of 6 SLOs defined for this serice (2 docs * 3 SLOs-per-doc = 6; 2 SLOs named 'latency', 2 SLOs named 'errors', and 2 SLOs named 'availability'). However, the 'ocm' [visual report page](https://visual-app-interface.devshift.net/reports#/reports/ocm/2021-08-01.yml) only renders 3 SLOs. **This is NOT working as intended.** The tools do not take into account the possibility of a service defining the same SLO name multiple times across multiple docs for the same service. Presumably the SLO datapoints in dash.db are incorrectly overwriting one another or are incorrectly being aggregated together.
+
+The service ['service-registry'](https://gitlab.cee.redhat.com/service/app-interface/-/tree/6762ce9ae3c34abedc21abf768437a5e7753652a/data/services/service-registry/slo-documents) would have [a similar problem](https://visual-app-interface.devshift.net/reports#/reports/service-registry/2021-08-01.yml)(although its visual report page fails to render for a seemingly-unrelated-reason) to 'ocm'.
+
+## Proposal
+
+### Overview
+
+The 'name' ([example](https://gitlab.cee.redhat.com/service/app-interface/-/blob/8bba50902109207d7e8a0b8f856bec92ede1e482/data/services/ocm/slo-documents/accounts-mgmt.yml#L7)) of these SLO docs will be introduced as an identifier for the SLO metric data stored in dashdotdb.
+
+SLO data will be stored in report-YML files in the following format:
+```
+service_slo:
+  slo_doc_name1:
+  - cluster: app-sre-prod-01
+    namespace: app-interface-production
+    slo_name: Reconciliation time of app-interface integrations
+    slo_value: 27.0
+    slo_target: 95.0
+  slo_doc_name2:
+  - ...
+```
+
+Visual-Qontract will have a new column `Doc Name` in the 'Service SLO' table.
+
+### Schema Changes
+
+Here are three proposals for how a 'slo_doc_name' identifier can be introduced to the schema:
+* [Option 1] (least effort): Update the 'ServiceSLO' table to include a 'SLODocName' column of type 'string'.
+* [Option 2] (author's recommendation): Update the 'ServiceSLO' table to include a 'SLODocName' column of type 'integer', which is a foreign key to a new table 'SLODocName'. Table 'SLODocName' contains 'id' (primary key, integer), and 'name' (string, unique)
```

Comment:
I am unsure of the value brought by externalizing the doc name in it own table here. If this table contains only an id and a name, and is only foreign-keyed from the ServiceSLO table, it would be simpler and easier to manage to do the option 1.

Reply:
Also, if we go for this option, the new table name should probably be more `SLODoc` rather than `SLODocName`

Reply:
'Option 1' is certainly easier. The externalized table was a bit of a future-proofing exercise, but quite happy to plan for 'Option 1' if we don't think that's necessary.

Otherwise, 'SLODoc' would probably be a more appropriate table name, sure.

Reply:
I do agree that we could do it with Option 1, but definitely is more in line with the overall schema design to put it in its own table, especially since we can have multiple SLO definitions relating to one SLO document, so I agree with Option 2 in this case. DB normalization rules are always somewhat obscure to me, but I guess that should be to make it satisfy the 2NF.

Reply:
Unless further discussion arises, I'll plan on utilizing 'Option 2' for this.